### PR TITLE
Add navigation overlay block

### PR DESCRIPTION
**Pull request by scruffian**
## What?
Adds a block that wraps the navigation overlay.

## Why?
This will be useful for two reasons:

1. It allows us to style the navigation overlay using theme.json
2. It will be useful in allowing us to customize the navigation overlay as per https://github.com/WordPress/gutenberg/issues/43852

## How?
Creates a new block called Navigation Overlay, which is used as the navigation overlay.

## Testing Instructions
1. Switch to a block theme
3. Add this to the theme.json for your theme:
```
			"core/navigation-overlay": {
				"color": {
					"background": "blue",
					"text": "#00ff00"
				}
			},
```
4. Add a navigation block and set it to collapse (the default setting)
Screenshot 2023-10-24 at 11 51 42
5. Open the navigation overlay and confirm that the colors from the theme.json are applied:

Screenshot 2023-10-24 at 11 52 24

### CODE REVIEW

**Comment by bgardner** [comment]
I love this concept, and it would help me offload stuff in style.css in various themes.

File: lib/blocks.php [hunk 1]
```diff
@@ -31,6 +31,7 @@ function gutenberg_reregister_core_block_types() {
 				'media-text',
 				'missing',
 				'more',
+				'navigation-overlay',
```
Comment (scruffian) [inline lib/blocks.php:34]: This is needed so that Global Styles outputs the styles

File: packages/block-library/src/navigation-overlay/index.php
Comment (MaggieCabrera) [inline packages/block-library/src/navigation-overlay/index.php:1]: Do we need this?

**Comment by scruffian** [comment]
This isn't working correctly, as the block itself isn't being output, so the CSS from Global Styles is never enqueued....

**Comment by richtabor** [comment]
Perhaps this effort should be considered as an experiment? 

I'm not 100% certain is this is the direction to take yet, but it's certainly worth exploring.

**Comment by scruffian** [comment]
I don't think we'll need this now...

**Pull request closed without merging**